### ISSUE TITLE
[Bug]: Pathless layout route causes invalid conflict

### ISSUE THREAD

**Issue by Ehesp** [remix-run/remix#1549]
### Which Remix packages are impacted?

- [ ] `remix` (Remix core)
- [ ] `create-remix`
- [ ] `@remix-run/architect`
- [ ] `@remix-run/cloudflare-workers`
- [X] `@remix-run/dev`
- [ ] `@remix-run/express`
- [ ] `@remix-run/netlify`
- [ ] `@remix-run/node`
- [ ] `@remix-run/react`
- [ ] `@remix-run/serve`
- [ ] `@remix-run/server-runtime`
- [ ] `@remix-run/vercel`

### What version of Remix are you using?

1.1.1

### What version of Node are you using? Minimum supported version is 14.

16.1.0

### Steps to Reproduce

Create a route structure like so:

```
routes/__foo/index.tsx
routes/__bar/bar.tsx
routes/__foo.tsx
routes/__bar.tsx
```

### Expected Behavior

That the`/` route goes via `__foo/index.tsx` & the `/bar` route goes via `__bar/bar.tsx`.

### Actual Behavior

The error is thrown when starting remix:

```
Path undefined defined by route "routes/__bar" conflicts with route "routes/__foo"
```

[You can see a reproduction here](https://codesandbox.io/s/reverent-kepler-89csf).

At the moment, if I edit the [check here](https://github.com/remix-run/remix/blob/main/packages/remix-dev/config/routesConvention.ts#L73) I can get it to work.

```diff
- if (typeof uniqueRouteId !== "undefined") {
+ if(uniqueRouteId)
```

The `uniqueRouteId` variable will never be undefined, since it's concatenated - this allows invalid checking of the unique route ids for the layout route.

```js
let uniqueRouteId = (fullPath || "") + (isIndexRoute ? "?index" : "");
```

**Comment by elkevinwolf** [comment]
Came here for the same issue. Apparently it got fixed in `v1.1.2` [here](https://github.com/remix-run/remix/pull/1347)

**Comment by Ehesp** [comment]
Looks like it - will close.
